src/webots/gui/WbSimulationView.cpp

```diff
@@ -232,10 +232,17 @@ QToolBar *WbSimulationView::createToolBar() {
   mToolBar->widgetForAction(action)->setObjectName("invisibleButton");
   mToolBar->widgetForAction(action)->setVisible("false");
 
+  WbActionManager::instance()->updateRenderingButton();
+  connect(WbSimulationState::instance(), &WbSimulationState::renderingStateChanged, this, &WbSimulationView::updateRendering);
+
```

Comment:
I would move these lines just after adding the `WbAction::RENDERING` to the toolbar